client/state/reader/site-blocks/reducer.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_SITE_BLOCK_REQUEST_SUCCESS,
+	READER_SITE_UNBLOCK_REQUEST_SUCCESS,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+
+/**
+ * Tracks all known block statuses, indexed by site ID.
+ *
+ * @param  {Array} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Array}        Updated state
+ */
+export function items( state = {}, action ) {
+	switch ( action.type ) {
+		case READER_SITE_BLOCK_REQUEST_SUCCESS:
+			return {
+				...state,
+				[ action.siteId ]: action.data.success
```

Comment:
Some of the reasons for failure could readily be covered by a 4xx (missing blog ID, for example) but less straightforward ones include:

- attempted to block site, but user has already blocked site. 
- attempted to unblock site, but the user doesn't have a current block for that site.

We get `success: false` in both cases. As we don't maintain the full list of the user's site blocks in Redux (there's no endpoint to list all their blocks), both these scenarios are possible if the user is blocking and unblocking in multiple tabs, for example.